### PR TITLE
HCA submission failure email

### PR DESCRIPTION
**Pull request by jholton**

### CODE REVIEW

File: app/models/health_care_application.rb [hunk 1]
```diff
@@ -12,10 +12,16 @@ class HealthCareApplication < ActiveRecord::Base
   validates(:state, presence: true, inclusion: %w[success error failed pending])
   validates(:form_submission_id_string, :timestamp, presence: true, if: :success?)
 
+  after_save :send_failure_mail, if: proc { |hca| hca.state_changed? && hca.failed? && hca.parsed_form&.dig('email') }
```
Comment (jholton) [inline app/models/health_care_application.rb]: Is this criteria accurate?

File: app/workers/hca/submission_job.rb [hunk 2]
```diff
@@ -4,19 +4,23 @@ module HCA
   class SubmissionJob
     include Sidekiq::Worker
 
+    sidekiq_options unique_for: 30.minutes
```
Comment (jholton) [inline app/workers/hca/submission_job.rb]: Good idea to add this?

File: app/workers/hca/submission_job.rb [hunk 3]
```diff
@@ -4,19 +4,23 @@ module HCA
...
+      health_care_application.update_attributes!(state: 'failed', form: msg['args'][1].to_json)
     end
 
     def perform(user_uuid, form, health_care_application_id)
       health_care_application = HealthCareApplication.find(health_care_application_id)
       user = User.find(user_uuid)
 
       begin
+        # TODO:  the following line is temporary, for testing
+        raise HCA::SOAPParser::ValidationError if form[:email] =~ /oddball|adhocteam/
```
Comment (lihanli) [inline app/workers/hca/submission_job.rb]: this should be `form['email']`

File: app/workers/hca/submission_job.rb (hunk 2 above)
Comment (jholton) [reply app/workers/hca/submission_job.rb]: Ah, didn't realize the bug was related specifically to `scheduler`.

Comment (jholton) [reply app/workers/hca/submission_job.rb]: Speaking of `sidekiq_options`, do we limit Sidekiq retries at all?  If not, potentially this job could fail for ~21 days before it gives up and sends the failure email.

**Pull request merged**